h1b/src/hil/flash/driver.rs

```diff
@@ -65,18 +77,24 @@ impl<'d, A: Alarm, H: Hardware> super::flash::Flash<'d> for FlashImpl<'d, A, H>
         self.hw.read(word)
     }
 
-    fn write(&self, target: usize, data: &[u32]) -> ReturnCode {
-        if data.len() > 32 { return ReturnCode::ESIZE; }
-        if self.program_in_progress() { return ReturnCode::EBUSY; }
+    fn write(&self, target: usize, data: &'d mut [u32]) -> (ReturnCode, Option<&'d mut [u32]>) {
+        let write_len = cmp::min(data.len(), MAX_WRITE_SIZE);
+
+        //if data.len() > 32 { return (ReturnCode::ESIZE, Some(data)); }
```

Comment:
We should delete this line instead of commenting it out, as the semantics of write have changed (and not temporarily).

I won't block this PR on this concern, though.